crates/pop-cli/src/commands/call/contract.rs

```diff
@@ -148,11 +158,21 @@ impl CallContractCommand {
 
 	/// If the contract has not been built, build it in release mode.
 	async fn ensure_contract_built(&self, cli: &mut impl Cli) -> Result<()> {
+		let path_flag = self.path.clone();
+		let project_path = if let Some(ref path_pos) = self.path_pos {
+			Some(path_pos) // Use positional path if present
+		} else {
+			path_flag.as_ref() // Otherwise, use the named path
+		};
 		// Build the contract in release mode
 		cli.warning("NOTE: contract has not yet been built.")?;
 		let spinner = spinner();
 		spinner.start("Building contract in RELEASE mode...");
-		let result = match build_smart_contract(self.path.as_deref(), true, Verbosity::Quiet) {
+		let result = match build_smart_contract(
+			project_path.as_deref().map(|v| &**v),
```

Comment:
This feels unnecesary, maybe you can change the code above to get an Option<Path>:
```rs
let path_flag = args.path.clone();
let project_path = if let Some(ref path_pos) = args.path_pos {
	Some(path_pos.as_path()) // Use positional path if present
} else {
	path_flag.as_deref() // Otherwise, use the named path
};

#[cfg(feature = "contract")]
if pop_contracts::is_supported(project_path)? 
```